<?xml version="1.0" encoding="UTF-8" ?>
<launch>
  <!-- arguments -->
  <arg name="nodelet_manager" default="velodyne_nodelet_manager" />
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen"/>

  <node pkg="nodelet" type="nodelet" name="hdl_people_detection_nodelet" args="load hdl_people_tracking/HdlPeopleDetectionNodelet $(arg nodelet_manager)">
  <rosparam file="$(find hdl_people_tracking)/config/hdl_people_detection.yaml" command="load" />
    <!--param name="static_sensor" value="true"/>
    <param name="downsample_resolution" value="0.1"/>
    <param name="pointcloud2_topic" value="/lslidar_c16/lslidar_point_cloud"/>
    <back subtraction params >
    <param name="backsub_resolution" value="0.2"/>
    <param name="backsub_occupancy_thresh" value="2"/-->
    <!-- clustering params>
    <param name="marcel_threshold" value="0.45"/>
    <param name="cluster_min_pts" value="10"/>
    <param name="cluster_max_pts" value="2048"/>
    <param name="cluster_min_size_x" value="0.2"/>
    <param name="cluster_min_size_x" value="0.15"/>
    <param name="cluster_min_size_y" value="0.2"/>
    <param name="cluster_min_size_y" value="0.1"/>
    <param name="cluster_min_size_z" value="0.3"/>
    <param name="cluster_min_size_z" value="0.15"/>
    <param name="cluster_max_size_x" value="1.0"/>
    <param name="cluster_max_size_y" value="1.0"/>
    <param name="cluster_max_size_z" value="2.0"/>
    <classification params>
    <param name="enable_classification" value="false"/-->
  </node>

  <node pkg="nodelet" type="nodelet" name="hdl_people_tracking_nodelet" args="load hdl_people_tracking/HdlPeopleTrackingNodelet $(arg nodelet_manager)">
    <rosparam file="$(find hdl_people_tracking)/config/hdl_people_tracking.yaml" command="load" />
    <!-- tracks with larger covariance matrix trace than this value will be removed -->
    <!--param name="remove_trace_thresh" value="1.0"/-->
    <!--param name="remove_time_thresh" value="5.0"/>
    <param name="human_radius" value="0.4"/-->
  </node>
  <node pkg="hdl_people_tracking" type="velocity_commands" name="velocity_commands" output="screen">
    <rosparam file="$(find hdl_people_tracking)/config/velocity_commands.yaml" command="load" />
  </node>

</launch>
